src/mlpack/methods/ann/layer/celu.hpp

```diff
@@ -0,0 +1,199 @@
+/**
+ * @file celu.hpp
+ * @author Gaurav Singh
+ *
+ * Definition of the CELU activation function as described by Jonathan T. Barron.
+ *
+ * For more information, read the following paper.
+ *
+ * @code
+ * @article{
+ *   author  = {Jonathan T. Barron},
+ *   title   = {Continuously Differentiable Exponential Linear Units},
+ *   year    = {2017}
+ * }
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_CELU_HPP
+#define MLPACK_METHODS_ANN_LAYER_CELU_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The CELU activation function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) &=& \left\{
+ *   \begin{array}{lr}
+ *    x & : x \ge 0 \\
+ *    \alpha(e^(\frac{x}{\alpha}) - 1) & : x < 0
+ *   \end{array}
+ * \right. \\
+ * f'(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     1 & : x \ge 0 \\
+ *     (\frac{f(x)}{\alpha}) + 1 & : x < 0
+ *   \end{array}
+ * \right.
+ * @f}
+ *
+ * In the deterministic mode, there is no computation of the derivative.
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class CELU
+{
+ public:
+  /**
+   * Create the CELU object using the specified parameter. The non zero
+   * gradient for negative inputs can be adjusted by specifying the CELU
+   * hyperparameter alpha (alpha > 0).
+   *
+   * @param alpha Scale parameter for the negative factor (default=1.0).
```

Comment:
Hm, I think we use `Default: 1.0` or `default = 1.0` in some other places, not sure, I think either is fine.